src/Bicep.Core/Semantics/Namespaces/AzNamespaceType.cs

```diff
@@ -298,157 +299,188 @@ private static IEnumerable<FunctionOverload> GetAzOverloads(ResourceScope resour
-                .Build();
-
-            // TODO: Change 'Full' to literal type after verifying in the runtime source
-            yield return new FunctionOverloadBuilder("reference")
-                .WithReturnType(LanguageConstants.Object)
-                .WithGenericDescription("Returns an object representing a resource's runtime state.")
-                .WithRequiredParameter("resourceNameOrIdentifier", LanguageConstants.String, "Name or unique identifier of a resource. When referencing a resource in the current template, provide only the resource name as a parameter. When referencing a previously deployed resource or when the name of the resource is ambiguous, provide the resource ID.")
-                .WithOptionalParameter("apiVersion", LanguageConstants.String, "API version of the specified resource. This parameter is required when the resource isn't provisioned within same template.")
-                .WithOptionalParameter("full", LanguageConstants.String, "Value that specifies whether to return the full resource object. If you don't specify 'Full', only the properties object of the resource is returned. The full object includes values such as the resource ID and location.")
-                .WithFlags(FunctionFlags.RequiresInlining)
-                .Build();
-
-            // TODO: Doc parameters need an update
-            yield return new FunctionWildcardOverloadBuilder("list*", AzConstants.ListWildcardFunctionRegex)
-                .WithReturnType(LanguageConstants.Any)
-                .WithGenericDescription("The syntax for this function varies by name of the list operations. Each implementation returns values for the resource type that supports a list operation. The operation name must start with list. Some common usages are `listKeys`, `listKeyValue`, and `listSecrets`.")
-                .WithRequiredParameter("resourceNameOrIdentifier", LanguageConstants.String, "Name or unique identifier of a resource. When referencing a resource in the current template, provide only the resource name as a parameter. When referencing a previously deployed resource or when the name of the resource is ambiguous, provide the resource ID.")
-                .WithRequiredParameter("apiVersion", LanguageConstants.String, "API version of resource runtime state. Typically, in the format, yyyy-mm-dd.")
-                .WithOptionalParameter("functionValues", LanguageConstants.Object, "An object that has values for the function. Only provide this object for functions that support receiving an object with parameter values, such as listAccountSas on a storage account. An example of passing function values is shown in this article.")
-                .WithFlags(FunctionFlags.RequiresInlining)
-                .Build();
+            if (sourceFileKind == BicepSourceFileKind.ParamsFile)
+            {
+                yield return new FunctionOverloadBuilder("getSecret")
```

Comment:
Could we make this consistent with the existing `getSecret` function? 
https://github.com/Azure/bicep/blob/eb4486b4a5370cfc531df72618e4e9f749a3aa6e/src/Bicep.Core/TypeSystem/Az/AzResourceTypeProvider.cs#L383-L385

Reply:
thanks for the comment, I just created a commit, function calls are similar now 👍